### PR TITLE
Harden redirector against redis failure

### PR DESCRIPTION
**Pull request by benhutton**

### CODE REVIEW

File: lib/wakes/middleware/redirector.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ def calculate_target_and_redirect(request)
         if target = TargetCalculator.new(request).run
           redirect(request.fullpath, target)
         end
+      rescue StandardError
```
Comment (joeosburn) [inline lib/wakes/middleware/redirector.rb:20]: Why not catch a more specific error here?

Comment (benhutton) [reply lib/wakes/middleware/redirector.rb:20]: I can go either way on that. My thinking was: are there errors that we _wouldn't_ want to catch and switch to this behavior on?

Comment (joeosburn) [reply lib/wakes/middleware/redirector.rb:20]: I think errors that we'd want to know about/surface. If this were part of the app, I'd say just report in sentry and leave it at that. Is there something we can do in wakes?

Comment (benhutton) [reply lib/wakes/middleware/redirector.rb:20]: mmm good point. not really. it does not have sentry integration. I will make the error more specific

File: lib/wakes/middleware/redirector.rb [hunk 2]
```diff
@@ -17,6 +17,8 @@ def calculate_target_and_redirect(request)
         if target = TargetCalculator.new(request).run
           redirect(request.fullpath, target)
         end
+      rescue Redis::BaseConnectionError
```
[the remark this answers is not in the document]
Comment (joeosburn) [reply lib/wakes/middleware/redirector.rb:20]: Thanks

Comment (gaganawhad) [inline lib/wakes/middleware/redirector.rb:20]: :heart:

**Review by gaganawhad — APPROVED** [review]

**Pull request merged**